scripts/msmtpq/msmtpq

```diff
@@ -56,35 +56,37 @@ err() { dsp '' "$@" '' ; exit 1 ; }
 ## ======================================================================================
 ##
 ## only if necessary (in unusual circumstances - e.g. embedded systems),
-##   enter the location of the msmtp executable  (no quotes !!)
-##   e.g. ( MSMTP=/path/to/msmtp )
-##   and uncomment the test for its existence
-MSMTP=msmtp
-#[ -x "$MSMTP" ] || \
-#  log -e 1 "msmtpq : can't find the msmtp executable [ $MSMTP ]"   # if not found - complain ; quit
+##   export the location of the msmtp executable before running this script  (no quotes !!)
+##   e.g. ( export MSMTP=/path/to/msmtp )
+if [ "$MSMTP" = "" ] ; then  # If MSMTP is unset or empty...
```

Comment:
MSMTP=${MSMTP:-""}
if [ -x "$MSMTP" ]; then
.....

Reply:
The check for the value of the variable (and setting it) is separate from checking if it's executable.

Reply:
> ```bash
> MSMTP=${MSMTP:-""}
> ```

That wouldn’t be correct. The default value of `MSMTP` should be “msmtp” not an empty string. I could change it to

```bash
MSMTP=${MSMTP:-msmtp}
```

but then the if statement would be less elegant:

```bash
MSMTP=${MSMTP:-msmtp}
if [ "$MSMTP" != "msmtp" ] && [ ! -x "$MSMTP" ] ; then
  log -e 1 "msmtpq : can't find the msmtp executable [ $MSMTP ]"   # if not found - complain ; quit
fi
```

> The check for the value of the variable (and setting it) is separate from checking if it's executable.

Yes, but the test was only supposed to be run if you changed the value of `MSMTP`. Before this change, there was a comment that said:

```bash
##   enter the location of the msmtp executable  (no quotes !!)
##   e.g. ( MSMTP=/path/to/msmtp )
##   and uncomment the test for its existence
```

If that test got run with `MSMTP`’s default value, then you would end up with:

```bash
[ -x msmtp ]
```

which will fail since [`-x` expects path to a file](https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html), not a command.